File: back/src/main/java/com/realive/dto/order/ProductQuantityDTO.java
package com.realive.dto.order;

import jakarta.validation.constraints.Min; // 추가
import jakarta.validation.constraints.NotNull; // 추가
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductQuantityDTO {
    @NotNull(message = "상품 ID는 필수입니다.") // 상품 ID는 null이 아니어야 함
    private Long productId;
    @Min(value = 1, message = "수량은 최소 1개 이상이어야 합니다.") // 수량은 1개 이상이어야 함
    private int quantity;
}

File: front/src/types/orders/productQuantityDTO.ts

export interface ProductQuantityDTO {
    productId: number; // @NotNull -> 필수 필드, Spring의 Long은 TypeScript에서 number로 매핑됩니다.
    quantity: number;  // @Min(value = 1) -> 최소 1 이상, Spring의 int는 TypeScript에서 number로 매핑됩니다.
}